### ISSUE TITLE
[Bug][a11y]: Download test file cannot be selected

### ISSUE THREAD

**Issue by SoftVision-LorandJanos** [mozilla-mobile/fenix#27315]
### Steps to reproduce

1. Turn TalkBack (or the available screen reader) on.
2. Go to thinkbroadband.com/download and tap on one of the items.
3. Tap on any of the items.

### Expected behaviour

- The item is selected.
- Talkback reads "5MB, link. Double tap to activate".

### Actual behaviour

- The item is not selected with direct tap.
- When selected by swipe, Talkback reads "link. Double tap to activate"

### Device name

Google Pixel 4

### Android version

Android 12

### Firefox release type

Firefox Nightly

### Firefox version

107.0a1

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-21750)

**Comment by Mugurell** [comment]
There is indeed a difference in the a11y structure seen in Chrome versus Fenix:

| Chrome | Fenix Nightly |
| --- | --- |
| <video src="https://user-images.githubusercontent.com/11428869/194504061-56f193ff-5efd-4d87-b831-61673f45dec4.mp4" /> | <video src="https://user-images.githubusercontent.com/11428869/194504232-9b81d8f8-06d9-480e-8e69-a47055b90016.mp4" />

So Chrome will have Talkback say "5MB, link. Double tap to activate"
While on Fenix Talkback will say "link. Double tap to activate" - the "5MB" is not read.

Seems like Chrome will though have issues downloading the file while on Fenix downloads work.
One important issue is that on Fenix as opposed to Chrome the "5MB.." element cannot be selected - tapped directly.

@SoftVision-LorandJanos Are these the issues you were seeing?

**Comment by SoftVision-LorandJanos** [comment]
@Mugurell yes, I've updated the ticket with the missing information. Thank you!

**Comment by Mugurell** [comment]
Since this issue happens in a webpage I'll move the ticket to Bugzilla for more investigation.

**Comment by Mugurell** [comment]
Moved to bugzilla: https://bugzilla.mozilla.org/show_bug.cgi?id=1794422

Change performed by the [Move to Bugzilla add-on](https://addons.mozilla.org/en-US/firefox/addon/move-to-bugzilla/).
